src/cryptography/hazmat/backends/openssl/aead.py

```diff
@@ -74,6 +86,9 @@ def _encrypt(backend, cipher_name, key, nonce, data, associated_data,
     ctx = _aead_setup(
         backend, cipher_name, key, nonce, None, tag_length, _ENCRYPT
     )
+    # CCM requires us to pass the length of the data before processing anything
+    if cipher_name[-3:] == b"ccm":
```

Comment:
I don't like checking on this string. Let's think of a better API.

Can we just call `_set_length` for all AEADs?